src/device-registry/models/Event.js

```diff
@@ -492,6 +497,12 @@ eventSchema.statics = {
       delete search["recent"];
       delete search["page"];
 
+      /**
```

Comment:
Just some brief notes for the entity logic below....added since the logic can be intimidating even to the author...it is rather too much code. :) 

We shall refactor soon....